packages/frontend-2/components/workspace/ProjectList.vue

```diff
@@ -0,0 +1,246 @@
+          New project
+        </FormButton>
+      </div>
+    </div>
+    <CommonLoadingBar :loading="showLoadingBar" class="my-2" />
+    <ProjectsDashboardEmptyState
+      v-if="showEmptyState"
+      @create-project="openNewProject = true"
+    />
+    <template v-else-if="projects?.items?.length">
+      <ProjectsDashboardFilled :projects="projects" />
+      <InfiniteLoading
+        :settings="{ identifier: infiniteLoaderId }"
+        @infinite="infiniteLoad"
+      />
+    </template>
+    <CommonEmptySearchState v-else-if="!showLoadingBar" @clear-search="clearSearch" />
+    <ProjectsAddDialog v-model:open="openNewProject" :workspace-id="workspaceId" />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { MagnifyingGlassIcon, Squares2X2Icon } from '@heroicons/vue/24/outline'
+import {
+  useApolloClient,
+  useQuery,
+  useQueryLoading,
+  useSubscription
+} from '@vue/apollo-composable'
+import { debounce } from 'lodash-es'
+import { graphql } from '~~/lib/common/generated/gql'
+import {
+  getCacheId,
+  evictObjectFields,
+  modifyObjectFields
+} from '~~/lib/common/helpers/graphql'
+import type { User, UserProjectsArgs } from '~~/lib/common/generated/gql/graphql'
+import { UserProjectsUpdatedMessageType } from '~~/lib/common/generated/gql/graphql'
+import { ToastNotificationType, useGlobalToast } from '~~/lib/common/composables/toast'
+import { projectRoute } from '~~/lib/common/helpers/route'
+import { useActiveUser } from '~~/lib/auth/composables/activeUser'
+import type { InfiniteLoaderState } from '~~/lib/global/helpers/components'
+import type { Nullable, Optional, StreamRoles } from '@speckle/shared'
+import { workspacePageQuery } from '~~/lib/workspaces/graphql/queries'
+
+const logger = useLogger()
+
+const infiniteLoaderId = ref('')
+const cursor = ref(null as Nullable<string>)
+const selectedRoles = ref(undefined as Optional<StreamRoles[]>)
+const search = ref('')
+const debouncedSearch = ref('')
+const openNewProject = ref(false)
+const showLoadingBar = ref(false)
+
+const { activeUser, isGuest } = useActiveUser()
+const { triggerNotification } = useGlobalToast()
+const areQueriesLoading = useQueryLoading()
+const apollo = useApolloClient().client
+
+const props = defineProps<{
+  workspaceId: string
+}>()
+
+const {
+  result: projectsPanelResult,
+  fetchMore: fetchMoreProjects,
+  onResult: onProjectsResult,
+  variables: projectsVariables
+} = useQuery(workspacePageQuery, () => ({
+  workspaceId: props.workspaceId,
+  filter: {
+    search: (debouncedSearch.value || '').trim() || null
+    // onlyWithRoles: selectedRoles.value?.length ? selectedRoles.value : null
+  }
+}))
+
+onProjectsResult((res) => {
+  const projectsData = res.data?.workspace?.projects
+  if (projectsData) {
+    cursor.value = projectsData.cursor || null
+    infiniteLoaderId.value = JSON.stringify(projectsVariables.value?.filter || {})
+  }
+})
+
+const { onResult: onUserProjectsUpdate } = useSubscription(
```

Comment:
1. This subscription and its document is duplicated here. Not only that but it has a duplicate operation name, which is not allowed for GQLGen.
2. This subscription relates to User.projects, not Workspace.projects, and thus it doesn't belong here

If this component remains, please clean it up and only leave in what's actually needed for it